### PR TITLE
Add simple octree generation regression test.

### PR DESCRIPTION
**Pull request by feuerste**
This is a simple regression test, which would have caught #209. Hopefully this catches more refactoring errors, too.

### CODE REVIEW

**Comment by feuerste** [comment]
@SirVer FYI

File: src/generation.rs [hunk 1]
```diff
@@ -453,3 +453,48 @@ pub fn build_octree(
...
+        };
+        let mut points = vec![default_point; 100_001];
+        points[100_000].position = Vector3::new(2.0, 0.0, 0.0);
+        let mut bounding_box = Aabb3::zero();
+        for point in &points {
+            bounding_box = bounding_box.grow(Point3::from_vec(point.position));
+        }
+        let pool = scoped_pool::Pool::new(10);
+        let tmp_dir = TempDir::new("octree").unwrap();
+        build_octree(&pool, tmp_dir, 1.0, bounding_box, Points { points });
```
Comment (tblaette) [inline src/generation.rs]: would be good to delete the temp dir when the test has finished

File: src/generation.rs [hunk 2]
```diff
@@ -453,3 +453,48 @@ pub fn build_octree(
...
+            for p in &self.points {
+                func(p);
+            }
+        }
+        fn size_hint(&self) -> Option<usize> {
+            Some(self.points.len())
+        }
+    }
+    #[test]
+    fn test_generation() {
```
Comment (tblaette) [inline src/generation.rs]: What does this test actually test? It seems like it fails only if `build_octree()` panics. Is that the intended test?

File: src/generation.rs (hunk 1 above)
Comment (feuerste) [reply src/generation.rs]: That's exactly what `TempDir` from the [`tempdir`](https://docs.rs/tempdir/0.3.7/tempdir/) crate is about :)

File: src/generation.rs (hunk 2 above)
Comment (feuerste) [reply src/generation.rs]: Yes, I've overseen this twice already during refactoring, so it was definitely time for a test.

**Comment by feuerste** [comment]
Thanks for your reviews, Tom and Holger!

**Comment by feuerste** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user feuerste. Merge queue now has a length of 1.

File: src/generation.rs (hunk 1 above)
Comment (tblaette) [reply src/generation.rs]: Ha! Cool, didn't know this! :-)

**Pull request merged**